test/lib/next-modes/next-dev.ts

```diff
@@ -174,6 +174,18 @@ export class NextDevInstance extends NextInstance {
               throw new Error('Server has not finished restarting.')
             }
           })
+        } else {
+          try {
+            await retry(async () => {
+              const cliOutput = this.cliOutput.slice(cliOutputLength)
+
+              if (!this.serverCompiledPattern.test(cliOutput)) {
+                throw new Error('Server has not finished restarting.')
+              }
+            }, 5000)
+          } catch (e) {
+            /** Fail silently because not all change will be reflected in the server output */
```

Comment:
Isn't this a bug then? This means that every test where a patch file doesn't result in the CLI output will now wait idly for 5s.
Can we instead always throw if retry times out and then add an option instead where the patchFile callsites can control this behavior so that we know which tests don't actually need to wait for the server to reflect the changes?

Reply:
Added more control of it at callsite: https://github.com/vercel/next.js/pull/72439